tests/test_diffusion.py

```diff
@@ -0,0 +1,188 @@
+
+    # Particle density at simulation start should be more or less uniform
+    DensityField = Field('temp', np.zeros((5, 5), dtype=np.float32),
+                         np.linspace(np.min(grid.U.lon), np.max(grid.U.lon), 5, dtype=np.float32),
+                         np.linspace(np.min(grid.U.lat), np.max(grid.U.lat), 5, dtype=np.float32))
+    StartDensity = diffusers.density(DensityField, area_scale=False, relative=True)
+    grid.add_field(Field(type + 'StartDensity', StartDensity,
+                         DensityField.lon,
+                         DensityField.lat))
+
+    diffuse = diffusers.Kernel(SaptiallyVaryingDiffusion2D) if type == 'true_diffusion' else diffusers.Kernel(LagrangianDiffusionNoCorrection)
+
+    diffusers.execute(diffusers.Kernel(UpdatePosition) + diffuse, endtime=grid.U.time[0]+timestep*steps, dt=timestep,
+                      output_file=diffusers.ParticleFile(name=args.output+type),
+                      interval=timestep, recovery={ErrorCode.ErrorOutOfBounds: Send2PreviousPoint})
+
+    EndDensity = diffusers.density(DensityField, area_scale=False, relative=True)
+    grid.add_field(Field(type+'FinalDensity', EndDensity,
+                         DensityField.lon,
+                         DensityField.lat))
+    grid.write(output_file)
+
+    print(type + ' start variation = %s' % np.var(StartDensity))
+    print(type + ' end variation = %s' % np.var(EndDensity))
+
+    return [StartDensity, EndDensity]
+
+
+if __name__ == "__main__":
+    p = ArgumentParser(description="""
+    Example of diffusion in a non-uniform diffusivity field""")
+    p.add_argument('mode', choices=('scipy', 'jit'), nargs='?', default='jit',
+                   help='Execution mode for performing RK4 computation')
+    p.add_argument('-p', '--particles', type=int, default=1000,
+                   help='Number of particles to advect')
+    p.add_argument('-t', '--timesteps', type=int, default=1000,
+                   help='Timesteps of one second to run simulation over')
+    p.add_argument('-o', '--output', type=str, default='diffusion_test',
+                   help='Output filename')
+    args = p.parse_args()
+
+    # Density fields of random_walkers should show accumulation in the low-diffusivity centre of the space
+    # Diffusers should spread out evenly regardless of this same low-diffusivity
+    densities1 = diffusion_test(args.mode, 'true_diffusion', args.particles, args.timesteps, args.output)
+
+    tol = 0.0001
+    assert np.abs(np.var(densities1[0]) - np.var(densities1[1])) < tol, \
+        'Variance in diffuser particle density across cells is significantly different from start to end of simulation!'
+
+    densities2 = diffusion_test(args.mode, 'brownian_motion', args.particles, args.timesteps, args.output)
+    assert np.abs(np.var(densities2[0]) - np.var(densities2[1])) > tol, \
+        'Variance in brownian motion particle density across cells is not significantly different from start to end of simulation!'
```

Comment:
These `asserts` still won't be executed by Travis, as the py.test doesn't run the `main`